recipes/libelf/all/CMakeLists.txt

```diff
@@ -1,7 +1,33 @@
-cmake_minimum_required(VERSION 2.8.11)
-project(cmake_wrapper)
+cmake_minimum_required(VERSION 3.4)
+project(elf C)
 
 include(conanbuildinfo.cmake)
 conan_basic_setup()
 
-add_subdirectory("source_subfolder")
+if(EXISTS "${PROJECT_SOURCE_DIR}/source_subfolder/lib/sys_elf.h.w32")
+    file(RENAME "${PROJECT_SOURCE_DIR}/source_subfolder/lib/sys_elf.h.w32" "${PROJECT_SOURCE_DIR}/source_subfolder/lib/sys_elf.h")
+    file(RENAME "${PROJECT_SOURCE_DIR}/source_subfolder/lib/config.h.w32" "${PROJECT_SOURCE_DIR}/source_subfolder/config.h")
+endif()
+
+file(GLOB_RECURSE SOURCES "source_subfolder/lib/*.c")
+file(GLOB_RECURSE HEADERS "source_subfolder/lib/*.h")
+
+add_library(${PROJECT_NAME} ${SOURCES})
+set_target_properties(${PROJECT_NAME} PROPERTIES PUBLIC_HEADER "${HEADERS}")
+target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC "source_subfolder/lib")
+target_compile_definitions(${PROJECT_NAME} PUBLIC HAVE_MEMCMP=1 HAVE_MEMCPY=1 HAVE_MEMMOVE=1)
+
+if(MSVC AND BUILD_SHARED_LIBS)
```

Comment:
```suggestion
if(WIN32AND BUILD_SHARED_LIBS)
```

Reply:
WINDOWS_EXPORT_ALL_SYMBOLS is ignored if not MSVC or not SHARED, so it doesn't really matter.

Reply:
I think this also works for mingw.
https://cmake.org/cmake/help/latest/prop_tgt/WINDOWS_EXPORT_ALL_SYMBOLS.html is talking about MS-compatible tools.

Reply:
MinGW is like others GNU tools: symbols are exported by default.

Reply:
No that's not true. You also need to export symbols with a declaration.
http://www.mingw.org/wiki/sampledll

Reply:
you mean global variables? WINDOWS_EXPORT_ALL_SYMBOLS doesn't export those symbols, manual export is still required.

Reply:
WINDOWS_EXPORT_ALL_SYMBOLS will export all non-static functions and classes.
For both msvc and mingw.
And indeed, manual export is required for global variables.

What I want to say is that it works the same for mingw and msvc.

Reply:
I've tested, and symbols are exported if MinGW and shared, without adding `WINDOWS_EXPORT_ALL_SYMBOLS`.